pkg/component/kubeapiserver/kube_apiserver.go

```diff
@@ -575,3 +578,45 @@ func getLabels() map[string]string {
 		v1beta1constants.LabelRole: v1beta1constants.LabelAPIServer,
 	}
 }
+
+// ComputeKubeAPIServerServiceAccountConfig computes the [ServiceAccountConfig]
+// needed to configure a kube-apiserver.
+func ComputeKubeAPIServerServiceAccountConfig(
+	config *gardencorev1beta1.ServiceAccountConfig,
+	externalHostname string,
+	serviceAccountKeyRotationPhase gardencorev1beta1.CredentialsRotationPhase,
+) ServiceAccountConfig {
+	var (
+		defaultIssuer = "https://" + externalHostname
+		out           = ServiceAccountConfig{
+			Issuer:        defaultIssuer,
+			RotationPhase: serviceAccountKeyRotationPhase,
+		}
+	)
+
+	if config == nil {
+		return out
+	}
+
+	out.ExtendTokenExpiration = config.ExtendTokenExpiration
+	out.MaxTokenExpiration = config.MaxTokenExpiration
+
+	if config.Issuer != nil {
+		out.Issuer = *config.Issuer
+	}
+	out.AcceptedIssuers = config.AcceptedIssuers
+	if out.Issuer != defaultIssuer && !slices.Contains(out.AcceptedIssuers, defaultIssuer) {
+		out.AcceptedIssuers = append(out.AcceptedIssuers, defaultIssuer)
+	}
+	if config.Issuer == nil {
+		// ensure defaultIssuer is not duplicated in the accepted issuers
+		for i, val := range out.AcceptedIssuers {
+			if val == defaultIssuer {
+				out.AcceptedIssuers = append(out.AcceptedIssuers[:i], out.AcceptedIssuers[i+1:]...)
+				break
```

Comment:
Should this cover the scenario when the issuer is the default issuer?
If yes, it should remove all occurances of default issuer in acceptedIssuers, not only the first one?

```suggestion
	if out.Issuer == defaultIssuer {
		// ensure defaultIssuer is not duplicated in the accepted issuers
		for i, val := range out.AcceptedIssuers {
			if val == defaultIssuer {
				out.AcceptedIssuers = append(out.AcceptedIssuers[:i], out.AcceptedIssuers[i+1:]...)
```

PS. I just saw this is moved from another package, still question might be valid.

Reply:
> it should remove all occurances of default issuer in acceptedIssuers, not only the first one?

Yes. We can probably do this in another PR.